src/Morphir/JsonSchema/Backend.elm

```diff
@@ -274,12 +274,22 @@ mapType typ =
 
         Type.Record _ fields ->
             let
-                requiredFields : List String
+                requiredFields : List FieldName
                 requiredFields =
                     fields
-                        |> List.map
+                        |> List.filterMap
                             (\field ->
-                                field.name |> Name.toCamelCase
+                                case field.tpe of
+                                    Type.Reference _ (( packageName, moduleName, localName ) as fQName) argTypes ->
+                                        case ( FQName.toString fQName, argTypes ) of
+                                            ( "Morphir.SDK:Maybe:maybe", _ ) ->
+                                                Nothing
+
+                                            _ ->
+                                                Just (field.name |> Name.toCamelCase)
+
+                                    _ ->
+                                        Just (field.name |> Name.toCamelCase)
```

Comment:
This looks good. As I mentioned though it's not enough. You also need to remove the `Maybe` wrapper from the field type because with this change we use field optionality to express the same thing.